M365/MDO/MDOThreatPolicyChecker.ps1

```diff
@@ -279,7 +288,12 @@ begin {
             $emailInRule = $emailExceptionInRule = $groupInRule = $groupExceptionInRule = $domainInRule = $domainExceptionInRule = $false
 
             if ($Outbound) {
-                Write-Verbose "Checking outbound rule $($rule.Name)"
+                $message = "Checking outbound spam rule: `"$($rule.Name)`""
+                if ($ShowDetailedExplanation) {
+                    Write-Host "`t`t$message"
+                } else {
+                    Write-Verbose $message
+                }
```

Comment:
Personally, instead of doing this same action over and over and over again, I would recommend making a function called `Write-DetailedExplanationOption` that would contain this `if` statement that would take the passed value message and then determine if we need to `Write-Host` or `Write-Verbose` depending on the `$ShowDetailedExplanation` value.

Reply:
By also doing it this way, it greatly reduces the lines of code within your script.